libgnucash/engine/gnc-pricedb.c

```diff
@@ -2577,21 +2585,22 @@ gnc_numeric
 gnc_pricedb_get_nearest_price (GNCPriceDB *pdb,
                                const gnc_commodity *orig_currency,
                                const gnc_commodity *new_currency,
-                               const time64 t)
+                               const time64 t,
+                               gboolean before_date)
```

Comment:
I'm not 100% sure it'll be nice to amend the signature of this function. As you notice, it affects custom reports. Wouldn't adding `gnc_pricedb_get_nearest_before_price` be a safer approach?

Reply:
Agree, I have created functions as suggested.